### PR TITLE
More efficiently validate NumericalDataSeries

### PR DESCRIPTION
**Pull request by nawatts**
`_validate` currently loops through the entire data array in an `Array.prototype.every` callback, repeating the same work `data.length` times.

### CODE REVIEW

File: lib/models/NumericalDataSeries.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import DataSeries from './DataSeries';
 
+const NUMERIC_KEYS = ['x_min', 'x_value', 'x_max', 'y_min', 'y_value', 'y_max'];
```
Comment (nickdoesdesign) [inline lib/models/NumericalDataSeries.js:3]: Great use of `const` here.

**Comment by nickdoesdesign** [comment]
+1 merge it.

**Review by choochburger — APPROVED** [review]
Looks good 👍

**Comment by alamb** [comment]
I am curious -- did  you do any measurements to show this code was slow / could be faster, or was this change based on code inspection?

**Comment by nawatts** [comment]
@alamb Just code inspection before changing. Since you mentioned it, I put together a quick benchmark. For validating a valid array of data points with `x_value` and `y_value` keys (times and speedup are average of 100 trials):

| # data points | old implementation | new implementation | speedup |
| --- | --- | --- | --- |
100 | 10.75 ms | 0.13 ms | 83x |
1000 | 1136 ms | 1.12 ms | 1014x |

**Comment by alamb** [comment]
Sounds sweet to me!

**Pull request merged**